drivers/pwm/pwm_nrfx.c

```diff
@@ -243,12 +242,11 @@ static int pwm_nrfx_set_cycles(const struct device *dev, uint32_t channel,
 static int pwm_nrfx_get_cycles_per_sec(const struct device *dev, uint32_t channel,
 				       uint64_t *cycles)
 {
-	/* TODO: Since this function might be removed, we will always return
-	 * 16MHz from this function and handle the conversion with prescaler,
-	 * etc, in the pin set function. See issue #6958.
-	 */
+#ifdef CONFIG_NRFX_PWM120
+	*cycles = 320ul * 1000ul * 1000ul;
+#else
 	*cycles = 16ul * 1000ul * 1000ul;
-
+#endif
```

Comment:
How about using this approach from UARTE:
https://github.com/zephyrproject-rtos/zephyr/commit/036a85b517deab873d60bf2eff552fb5e1a4afc9
https://github.com/zephyrproject-rtos/zephyr/commit/7c011741faecd60d482087c000666228064fb0af

Reply:
Since it is a hotfix, could be postpone the changes until the DMM rework?

Reply:
I don't see connection between this and DMM, but I'm fine with this staying temporarily if you post a follow-up PR.